### PR TITLE
Core/ts version task 310 orderbook service

### PR DESCRIPTION
**Pull request by WilliamHawkingStephen**
## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/domain/interfaces/math.interface.ts [hunk 1]
```diff
@@ -32,5 +31,20 @@ export namespace IMath {
     lt(param1: NumberEngine, param2: NumberEngine, scale?: number): boolean;
     gt(param1: NumberEngine, param2: NumberEngine, scale?: number): boolean;
     toFixed(param1: NumberEngine, param2: NumberEngine, scale?: number): string;
+    max(
+      param1: NumberEngine,
+      param2: NumberEngine,
+      params?: NumberEngine[],
+    ): NumberEngine;
+    floor(param: NumberEngine): NumberEngine;
+    max(
```
Comment (isaac-fibonacci) [inline ts-version/src/domain/interfaces/math.interface.ts:40]: The max function is defined twice

File: ts-version/src/domain/ports/repository/order.ts [hunk 2]
```diff
@@ -15,6 +15,16 @@ export namespace IOrderRepo {
     endTime?: Date;
   };
 
+  export type getAllInput = {
+    network: INetwork;
```
Comment (isaac-fibonacci) [inline ts-version/src/domain/ports/repository/order.ts:19]: Cannot find name 'INetwork'.

File: ts-version/src/presentation/http/controllers/order/process-order.adapter.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import Order from "@entity/order.entity";
+import { IProcessOrderService } from "@port/process-order-service.interface";
+import BaseController from "../base.controller";
```
Comment (isaac-fibonacci) [inline ts-version/src/presentation/http/controllers/order/process-order.adapter.ts:3]: fix import error

File: ts-version/src/presentation/http/controllers/order/process-order.adapter.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { PrivateRequest } from "@type/request.type";
```
Comment (isaac-fibonacci) [inline ts-version/src/presentation/http/controllers/order/process-order.adapter.ts:4]: fix import error

File: ts-version/src/presentation/http/controllers/order/process-order.adapter.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { Response } from "express";
+import INetwork from "@port/Network.interface";
```
Comment (isaac-fibonacci) [inline ts-version/src/presentation/http/controllers/order/process-order.adapter.ts:6]: fix import error

File: ts-version/src/domain/services/orderbook/orderbook.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    let orderbookRows: IOrderbookService.Step | object = {};
+    for (let i = 0; i < parsedOrders.length; i++) {
+      const order = parsedOrders[i];
+      const rawPrice = mathEngine.div(order.price, steps);
+      const price = mathEngine.floor(rawPrice);
+      const priceWithSteps = mathEngine.mul(price, steps);
+      orderbookRows = {
+        price: priceWithSteps,
+        amount: order.unfilled,
+        count: i,
```
Comment (isaac-fibonacci) [inline ts-version/src/domain/services/orderbook/orderbook.ts:65]: Argument of type '{}' is not assignable to parameter of type 'never'.ts(2345)

File: ts-version/src/domain/interfaces/math.interface.ts (hunk 1 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/domain/interfaces/math.interface.ts:40]: done

File: ts-version/src/domain/ports/repository/order.ts (hunk 2 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/domain/ports/repository/order.ts:19]: done

File: ts-version/src/presentation/http/controllers/order/process-order.adapter.ts (hunk 3 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/presentation/http/controllers/order/process-order.adapter.ts:3]: done

File: ts-version/src/presentation/http/controllers/order/process-order.adapter.ts (hunk 4 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/presentation/http/controllers/order/process-order.adapter.ts:4]: done

File: ts-version/src/domain/services/orderbook/orderbook.ts (hunk 6 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/domain/services/orderbook/orderbook.ts]: done

**Pull request merged**